### PR TITLE
cephfs: implement getSnapshotInfo with go-ceph

### PR DESCRIPTION
**Pull request by Madhu-1**
implement getsnapshotInfo function with go-ceph to get subvolume information.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: internal/cephfs/util.go [hunk 1]
```diff
@@ -149,17 +135,8 @@ func genSnapFromOptions(ctx context.Context, req *csi.CreateSnapshotRequest) (sn
...
-	layout := "2006-01-02 15:04:05.000000"
-	// TODO currently parsing of timestamp to time.ANSIC generate from ceph fs is failing
-	var t time.Time
-	t, err := time.Parse(layout, createTime)
-	if err != nil {
-		util.ErrorLog(ctx, "failed to parse time %s %v", createTime, err)
-		return tm, err
-	}
-	tm, err = ptypes.TimestampProto(t)
+func parseTime(ctx context.Context, createTime time.Time) (*timestamp.Timestamp, error) {
```
Comment (nixpanic) [inline internal/cephfs/util.go:138]: This function should probably have a unit test. No need to add it now, but maybe it should be reported as an Issue.

File: internal/cephfs/snapshot.go [hunk 2]
```diff
@@ -91,44 +94,31 @@ func deleteSnapshot(ctx context.Context, volOptions *volumeOptions, cr *util.Cre
 }
 
 type snapshotInfo struct {
-	CreatedAt        string `json:"created_at"`
+	CreatedAt        time.Time
 	CreationTime     *timestamp.Timestamp
-	DataPool         string `json:"data_pool"`
 	HasPendingClones string `json:"has_pending_clones"`
```
Comment (nixpanic) [inline internal/cephfs/snapshot.go:99]: Are the JSON attributes still needed?

**Review by nixpanic — CHANGES_REQUESTED** [review]
looks good, but maybe need to cleanup the JSON attributes? Also the name of the function in the subject of the commit and PR is not correctly capitalized.

File: internal/cephfs/snapshot.go (hunk 2 above)
Comment (Madhu-1) [reply internal/cephfs/snapshot.go]: Not required, removed it now

File: internal/cephfs/util.go (hunk 1 above)
Comment (Madhu-1) [reply internal/cephfs/util.go:138]: sure I will create an issue to ad unit testing for the functions which don't make any call to the ceph cluster.

**Review by nixpanic — COMMENTED** [review]
Change looks good, but now needs a manual rebase.

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
An other fix is needed:
```
internal/cephfs/controllerserver.go:586:97: `doSnapshot` - `cr` is unused (unparam)
func doSnapshot(ctx context.Context, volOpt *volumeOptions, subvolumeName, snapshotName string, cr *util.Credentials) (snapshotInfo, error) {
                                                                                                ^
```

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
Delpoying Vault failed again, #1756 and #1757 might help with fixing and identifying the issues.

**Comment by nixpanic** [comment]
@mergifyio rebase

**Pull request merged**